project/yxuan0329/proposal.rst

```diff
@@ -0,0 +1,171 @@
+more comprehensive and responsive mapping experience.
+
+
+
+Prospective Users
+=================
+
+.. Describe the users of your software and how they will use it.  It is OK to
+.. combine this section with the previous one (`Problem to solve`_).
+
+This project aims to provide to users that want to search multidimensional data 
+including spatial and temporal features.
+
+System Architecture
+===================
+
+.. Analyze how your system takes input, produces results, provide interface, and
+.. performs any other operations.  Describe the system's work flow.  You may
+.. consider to use a flow chart but it is not required.  Specify the constraints
+.. assumed in your system.  Describe the modularization of the system.
+(Will be finished soon after discussion)
+
+
+Implementation Plan
+===================
+
+1. Frontend design:
+The search engine will provide a simple UI with one text block for users to 
+enter query for searching, and one text block list for showing the searching 
+results.
+
+2. Datasets:
+Generate the database with about 50 map data entry by myself for validating the 
+R-tree algorithm can works correctly. 
+
+
+API Description
+===============
+
+.. Show how your system can be programmed.  You are supposed to implement the
+.. system using both C++ and Python.  Describe how a user writes a script in the
+.. system.
+(Will be finished soon after discussion)
+
+
+
+Engineering Infrastructure
+==========================
+
+.. Describe how you plan to put together the engineering system:
+
+.. 1. Automatic build system and how to build your program
+.. 2. Version control (show how you will use it)
+.. 3. Testing framework
+.. 4. Documentation
+
+.. Some of the above information should be included in the documentation in your
+.. software.
+
+.. You may use continuous integration, but it is not required.  If you use it,
+.. describe how it works in your code development.
+
+1. Automatic build system: CMake
+2. Version control: git
+3. Testing framework: Pytest
+4. Documentation: README.rst
+
+Schedule
+========
+
+.. Itemize the work to do and list them in a timeline.  Estimate the efforts of
+.. each item before the project starts.
+
+.. The schedule is expected to be adjusted during the development.  A schedule to
+.. accurately predict everything in the development is impossible and unnecessary.
+.. But the initial estimate is important, for the baseline becomes concrete when
+.. we adjust the plan.
+
+.. To start, you can list the to-do items on a weekly basis and assume the
+.. development of the project takes 8 weeks:
+
+* Planning phase (6 weeks from mm/dd to mm/dd):
+* Week 1 (10/30): Survey some reference about R-tree implementation
+* Week 2 (11/06): Create database with about 50-100 data
+* Week 3 (11/13): Implement R-tree with C++
+* Week 4 (11/20): Implement R-tree with C++
+* Week 5 (11/27): Implement front-end of the search engine
+* Week 6 (12/04): Improve the R-tree structure with temporal feature
```

Comment:
It is reasonable to use 4 weeks to implement R-tree.  I'd say your plan of time may be accurate.  In that case, you have only a week or two to add the temporal features to your system.  And you only have one or two weeks to make slides.  That does not sound like sufficient for making a good presentation.

I think you need to sacrifice something.

Reply:
I see. I will reschedule the plan and reserve more time for preparing the presentation.